File: Python/practical9.py
#Practical : 9- Write a python program to display above patterns.
#Pattern 1
# 1
# 1 2
# 1 2 3
# 1 2 3 4
# 1 2 3 4 5

#Pattern 2
# 1
# 2 3
# 4 5 6
# 7 8 9 10
# 11 12 13 14 15

#Pattern 3
#     *
#    * *
#   * * *
#  * * * *
# * * * * *

#Pattern 4
# A
# B B
# C C C
# D D D D
# E E E E E

#Pattern 1
print("Pattern 1:")
rows=5
for i in range(1, rows+1):
    for j in range(1, i+1):
        print(j, end=" ")
    print()
print()

#Pattern 2
print("Pattern 2:")
num=1
rows=5
k=2
for x in range(rows):
    for y in range(1,k):
        print(num, end=" ")
        num +=1
    print()
    k+=1
print()

#Pattern 3
print("Pattern 3:")
k=4
for i in range(0,5):
    for j in range(0,k):
        print(end=" ")
    k=k-1
    for j in range(0,i+1):
        print("*",end=" ")
    print()
print()

#Pattern 4
print("Pattern 4:")
num=65
for i in range(0,5):
    for i in range(0,i+1):
        c = chr(num)
        print(c, end=" ")
    num+=1
    print()
print()

File: Java/practical9.java
/*  PRACTICAL-9:Write a program that generate 6*6 two-dimensional matrix, filled
                with 0’s and 1’s , display the matrix, check every raw and column
                have an odd number’s of 1’s.
*/
// CODE:

import java.util.Scanner;
public class practical9 {
    public static int[][] create_fill_matrix() {
        int[][] matrix = new int[6][6];
        for (int i = 0; i < 6; i++) {
            for (int j = 0; j < 6; j++) {
                matrix[i][j] = (int)((Math.random() * 5) % 2);
            }
        }
        return matrix;
    }
    public static void displayMatrix(int[][] matrix) {
        System.out.print("\nThe Created Matrix is: \n");
        for (int i = 0; i < 6; i++) {
            for (int j = 0; j < 6; j++) {
                System.out.print(matrix[i][j] + " ");
            }
            System.out.println();
        }
    }
    public static void main(String[] args) {
        int my_matrix[][];
        int i, j, cnt;
        my_matrix = create_fill_matrix();
        displayMatrix(my_matrix);
        System.out.println("\nRows Having ODD no of 1's");
        for (i = 0; i < 6; i++) {
            cnt = 0;
            for (j = 0; j < 6; j++) {
                if (my_matrix[i][j] == 1) {
                    cnt++;
                }
            }
            if (cnt % 2 != 0) {
                System.out.println("Row-" + (i + 1) + " have 1 at the ODD No.");
            }
        }
        System.out.println("\nColumns Having ODD no of 1's");
        for (i = 0; i < 6; i++) {
            cnt = 0;
            for (j = 0; j < 6; j++) {
                if (my_matrix[j][i] == 1) {
                    cnt++;
                }
            }
            if (cnt % 2 != 0) {
                System.out.println("Column-" + (i + 1) + " have 1 at the ODD No.");
            }
        }
    }
}
